autohooks/template.py

```diff
@@ -25,6 +25,21 @@
 PYTHON3_SHEBANG = '/usr/bin/env python3'
 PIPENV_SHEBANG = '/usr/bin/env -S pipenv run python3'
 POETRY_SHEBANG = '/usr/bin/env -S poetry run python3'
+# For OS's that don't support '/usr/bin/env -S'.
+PIPENV_MULTILINE_SHEBANG = (
+    "/usr/bin/env sh\n"
```

Comment:
should by `/bin/sh`. as far as i know the multiline code works with all shells and `/bin/sh` is a symlink to the default shell of the system. therefore i don't see a reason for using `/usr/bin/env sh`. but maybe i am wrong.

Reply:
I am certain that you are correct that multi-line will work with `bash` or other shells.  I'll look into that some more.  Let me leave you with this thought: if it doesn't work with some odd system shell, what will be the failure mode? Will it be something obvious to the victim where then can submit a cogent bug report or will it be totally confusing and you get a bug report that says: _"Ahhh... it broked."

Reply:
_Wow!_ One can go years without knowing something right under one nose. To this day, I was certain that `/bin/sh` was universally, Bourne circa '77.  TL;DR: Though the temptation to use `/bin/dash`, I am sticking with your suspicion and with [this advice and converting to /bin/sh](https://askubuntu.com/a/141997).

Like you shared, `/bin/sh` has for a long time redirected to the _system_ shell (which doesn't infer the user's shell).  The shell that the core OS leverages for scripted tasks.

Historically, Linux linked `/bin/sh` --> bash`.  ubuntu broke with that and linked `/bin/sh` --> `dash`, a POSIX compliant, more efficient shell.  The primary reason is faster booting; not really too much value for user interactive scripting. macOS `/bin/sh` redirects (w/o using a link) to a 2007 version of `bash`, but recently, users are configured to us `zsh` due to licensing issues. So, I am expecting the `/bin/sh` to redirect to something else in the future.

macOS ships with `/bin/dash` so there is an urge to eek out performance using `/bin/dash`, but as the link in TD;DR points out, that just isn't recommended.